File: frontend/src/api/resume.ts
import client from './client'

export interface ExperienceEntry {
  title: string
  company: string
  location?: string
  start_date?: string
  end_date?: string
  description?: string
}

export interface EducationEntry {
  degree: string
  institution: string
  field?: string
  start_date?: string
  end_date?: string
}

export interface CertificationEntry {
  name: string
  issuer?: string
  date?: string
}

export interface ExtractedContent {
  skills: string[]
  languages: string[]
  experience: ExperienceEntry[]
  education: EducationEntry[]
  certifications: CertificationEntry[]
}

export interface Resume {
  id: string
  user_id: string
  file_name: string
  is_master: boolean
  version: number
  content_extracted: ExtractedContent | null
  created_at: string
  updated_at: string
}

export async function uploadResume(file: File): Promise<Resume> {
  const form = new FormData()
  form.append('file', file)
  const { data } = await client.post<Resume>('/api/v1/resumes', form, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
  return data
}

export async function getMyResume(): Promise<Resume> {
  const { data } = await client.get<Resume>('/api/v1/resumes/me')
  return data
}

export async function updateExtractedContent(
  payload: Partial<ExtractedContent>,
): Promise<Resume> {
  const { data } = await client.patch<Resume>('/api/v1/resumes/me', payload)
  return data
}

File: backend/app/schemas/resume.py
import uuid
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel


class ExperienceEntry(BaseModel):
    title: str
    company: str
    location: Optional[str] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    description: Optional[str] = None


class EducationEntry(BaseModel):
    degree: str
    institution: str
    field: Optional[str] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None


class CertificationEntry(BaseModel):
    name: str
    issuer: Optional[str] = None
    date: Optional[str] = None


class ExtractedContent(BaseModel):
    skills: list[str] = []
    languages: list[str] = []
    experience: list[ExperienceEntry] = []
    education: list[EducationEntry] = []
    certifications: list[CertificationEntry] = []


class UpdateExtractedContentRequest(BaseModel):
    skills: Optional[list[str]] = None
    languages: Optional[list[str]] = None
    experience: Optional[list[ExperienceEntry]] = None
    education: Optional[list[EducationEntry]] = None
    certifications: Optional[list[CertificationEntry]] = None


class ResumeResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    file_name: str
    is_master: bool
    version: int
    content_extracted: Optional[Any]
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
